From 7f724bbafbb1e170401dd5de201273ab8c8bc75f Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sun, 28 Aug 2022 14:24:02 -0700
Subject: [PATCH] fastcgi: Use value instead of address of sin6_port

This seems to be wrongly assigned where ipv4 sin_port is
equated to address of sin6_port and not value of sin6_port

Upstream-Status: Submitted [https://github.com/monkey/monkey/pull/375]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 plugins/fastcgi/fcgi_handler.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/fastcgi/fcgi_handler.c b/plugins/fastcgi/fcgi_handler.c
index 9e095e3c..e8e1eec1 100644
--- a/plugins/fastcgi/fcgi_handler.c
+++ b/plugins/fastcgi/fcgi_handler.c
@@ -245,7 +245,7 @@ static inline int fcgi_add_param_net(struct fcgi_handler *handler)
             struct sockaddr_in *s4 = (struct sockaddr_in *)&addr4;   
             memset(&addr4, 0, sizeof(addr4));
             addr4.sin_family = AF_INET;
-            addr4.sin_port = &s->sin6_port;
+            addr4.sin_port = s->sin6_port;
             memcpy(&addr4.sin_addr.s_addr, 
                    s->sin6_addr.s6_addr + 12, 
                    sizeof(addr4.sin_addr.s_addr));
-- 
2.37.2

